server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+        }
+        logger.debug("ImageId is: " + imageId);
+        return imageId;
+    }
+
+    /**
+     Finds the best InstanceType based on the values contained in the ComputeCapability.
+     If necessary the values are scaled upwards till they meet the requirement.
+
+     @param computeCapability The ComputeCapability to map.
+     @param distinction       A distinction string. Can either be "EC2" or "RDS".
+     @return A valid InstanceType / InstanceClass string.
+     @throws TransformationFailureException Gets thrown if the values numCpus and memSize are too big and there is no
+     valid InstanceType.
+     */
+    public String mapComputeCapabilityToInstanceType(ComputeCapability computeCapability, String distinction) throws
+        TransformationFailureException {
+        //TODO what to do with disk size?
+        Integer numCpus = computeCapability.getNumCpus().orElse(0);
+        Integer memSize = computeCapability.getMemSizeInMB().orElse(0);
+        //default type the smallest
+        final ImmutableList<InstanceType> instanceTypes;
+        if ("EC2".equals(distinction)) {
+            instanceTypes = EC2_INSTANCE_TYPES;
+        } else if ("RDS".equals(distinction)) {
+            instanceTypes = RDS_INSTANCE_CLASSES;
+        } else {
+            throw new IllegalArgumentException("Distinguisher not supported");
+        }
+        List<Integer> allNumCpus = instanceTypes.stream()
+            .map(InstanceType::getNumCpus)
+            .sorted()
+            .collect(Collectors.toList());
+        List<Integer> allMemSizes = instanceTypes.stream()
+            .map(InstanceType::getMemSize)
+            .sorted()
+            .collect(Collectors.toList());
+        // scale numCpus and memSize upwards if they are not represented in the lists
+        try {
+            logger.debug("Check numCpus: " + numCpus);
+            numCpus = checkValue(numCpus, allNumCpus);
+            logger.debug("Check memSize: " + memSize);
+            memSize = checkValue(memSize, allMemSizes);
+        } catch (IllegalArgumentException ie) {
+            String errorMessage = "Values numCpus: " + numCpus + " and/or memSize: " + memSize + " are to big. No " +
+                "InstanceType found";
+            logger.error(errorMessage);
+            throw new TransformationFailureException(errorMessage, ie);
+        }
+        //get instanceType from combination
+        String instanceType = findCombination(numCpus, memSize, instanceTypes, allNumCpus, allMemSizes);
+        logger.debug("InstanceType is: " + instanceType);
+        return instanceType;
+    }
+
+    /**
+     The value is taken from the Capability but turned into GB. The minimum is 20 GB the maximum 6144 GB
+
+     @param computeCapability The ComputeCapability to map.
+     @return An integer representing the diskSize that should be taken.
+     */
+    public Integer mapComputeCapabilityToDiskSize(ComputeCapability computeCapability) {
```

Comment:
If I understand this method properly, this converts the disk size to GB and caps it if it exceeds the min or max value right?
If so, I think the method name could better represent the functionality of the method.

I commented earlier about the TODO regarding disk_size. From what I've seen, this method currently only gets used for setting the AllocatedStorage of SQL nodes hosted on a Compute node. Since the min/max sizes are probably different for the EC2 instances, we should probably make it clear that this method is only used for the SQL nodes.